### PR TITLE
feat(Avatar): Implement Avatar Component

### PR DESCRIPTION
**Pull request by cs-paz**
closes https://linear.app/narmi/issue/NDS-638/%5Bavatar%5D-new-nds-component
Initial implementation of `Avatar` component

- Supports 3 sizes xs, s, m
- Supports an optional image background
- Supports an optional link. If you click the avatar button, it will navigate you to that link.

### CODE REVIEW

File: src/Avatar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  return (
+    <AsElement
+      elementType={linkurl ? "a" : "div"}
+      href={linkurl}
+      className={cc([
+        "nds-typography",
+        "nds-avatar",
+        `nds-avatar--${size}`,
+        "alignChild--center--center",
+        "bgColor--theme--primary",
```
Comment (cs-paz) [inline src/Avatar/index.tsx:38]: No requirement in profile to make the bgColor configurable. Would prefer to revisit this if the req ever comes up in a future feature

**Review by akdetrick — CHANGES_REQUESTED** [review]
Looks good!

Because this is a typescript component, we should also add it to the main `index.ts` to enable type hinting in consumers:

https://github.com/narmi/design_system/blob/7dbaba7e7ce5d993e44df3809fad8f7bd9732b00/src/index.ts#L2

File: src/Avatar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import cc from "classcat";
+import AsElement from "../util/AsElement";
+
+export interface AvatarProps {
+  /** aria-label for accessibility */
+  label: string;
+  /** Fixed height and width of the avatar. Default: "s". */
+  size?: "xs" | "s" | "m";
+  /** String to display in the avatar. If imgurl is provided, this will be ignored. */
+  initials?: string;
```
[the remark this answers is not in the document]
Comment (akdetrick) [reply src/Avatar/index.tsx:11]: A `kind` prop and the icon behavior can be added on later. Does not need to be included in the scope of this PR.

File: src/Avatar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  const backgroundImage = imgurl
+    ? { backgroundImage: `url(${imgurl})`, backgroundSize: "cover" }
+    : {};
+
+  return (
+    <AsElement
+      elementType={linkurl ? "a" : "div"}
+      href={linkurl}
+      className={cc([
+        "nds-typography",
```
[the remark this answers is not in the document]
Comment (cs-paz) [reply src/Avatar/index.tsx]: c4c63302

File: src/Avatar/index.scss [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+.nds-avatar--m {
+  height: 40px;
+  width: 40px;
+  font-size: var(--font-size-l);
+}
+
+.nds-avatar--l,
+.nds-avatar--xl {
+  @extend .nds-avatar--m;
+}
```
[the remark this answers is not in the document]
Comment (cs-paz) [reply src/Avatar/index.scss]: e799fad4

**Review by akdetrick — APPROVED** [review]

**Pull request merged**